File: soal4.js
const prompt = require("prompt-sync")({ sigint: true });

// Input data
let nama = prompt("Masukkan nama peserta: ");
let email = prompt("Masukkan email: ");
let divisi = prompt("Masukkan pilihan study group: ");
let jumlahInput = prompt("Masukkan jumlah peserta terdaftar saat ini: ");

// Ubah jumlah ke number
let jumlah = parseInt(jumlahInput);

// Validasi
let isValid = true;

// 1. Nama tidak boleh kosong
if (!nama || nama.trim() === "") {
  isValid = false;
}

// 2. Email tidak boleh kosong dan harus mengandung "@"
if (!email || email.trim() === "" || !email.includes("@")) {
  isValid = false;
}

// 3. Divisi tidak boleh kosong
if (!divisi || divisi.trim() === "") {
  isValid = false;
}

// 4. Jumlah harus angka dan minimal 1
if (isNaN(jumlah) || jumlah < 1) {
  isValid = false;
}

// Output hasil
if (isValid) {
  console.log("\nData valid");
} else {
  console.log("\nData tidak valid");
}

File: soal4.py
def validasi_data(nama, email, divisi, jumlah):
    # Cek nama
    if not nama.strip():
        return "Data tidak valid"

    # Cek email
    if not email.strip() or "@" not in email:
        return "Data tidak valid"

    # Cek pilihan divisi
    if not divisi.strip():
        return "Data tidak valid"

    # Cek jumlah peserta
    if not jumlah.isdigit() or int(jumlah) < 1:
        return "Data tidak valid"

    return "Data valid"


# Input dari user
nama = input("Masukkan nama peserta: ")
email = input("Masukkan email: ")
divisi = input("Masukkan pilihan study group: ")
jumlah = input("Masukkan jumlah peserta terdaftar saat ini: ")

# Panggil fungsi validasi
hasil = validasi_data(nama, email, divisi, jumlah)

print("Hasil:", hasil)
